### ISSUE TITLE
Issue with HTTP async result management

### ISSUE THREAD

**Issue by joowani** [arangodb/arangodb#2009]
Hello, I am using ArangoDB 3.0.5.

When I try to retrieve a result of an asychronous job using the [REST API](https://docs.arangodb.com/HTTP/AsyncResultsManagement/index.html), namely the **PUT /_api/job/{job-id}** endpoint, there does not seem to be a good way for me to tell whether a 404 (for example) was returned because the async job is missing, or the asynchrously executed task itself returned a 404. There seems to be loss of information here. At the moment I am forced to use a combination of ArangoDB's internal error number and message to figure out what happened. Is there a way to clearly determine where the status code is coming from?

Thanks,
Joohwan

**Comment by jsteemann** [comment]
As far as I can see, the server will return a 404 error for a job that does not exist, and an HTTP 204 for a job that is still in progress. 
If the server returns a 404 error, then it is indeed important to figure out whether the original job returned a 404 error or if the error is reported because the job was not found. 
I think this *can* be done by checking for the "x-arango-async-id" header in the server's response. This header will only be returned and present if the server response contains the original job's response.

That means calling **PUT /_api/job/{job-id}** for an unknown job will return HTTP 404 *without* the "x-arango-async-id" header, and it for a job that finished but itself returned an HTTP 404 will return 404 *with* the header.

Example following:
**Original request (not using async job API)**:
```
curl -X GET http://127.0.0.1:8529/_api/collection/does-not-exist --dump - --basic --user "root:"
HTTP/1.1 404 Not Found
Content-Type: application/json; charset=utf-8
Server: ArangoDB
Connection: Keep-Alive
Content-Length: 94

{"error":true,"code":404,"errorNum":1203,"errorMessage":"unknown collection 'does-not-exist'"}
```
The above call returns a 404.
Next the same operation will be executed using the **async job API**:
```
curl --header "x-arango-async: store" -X GET http://127.0.0.1:8529/_api/collection/does-not-exist --dump - --basic --user "root:"
HTTP/1.1 202 Accepted
X-Arango-Async-Id: 147185081573075
Server: ArangoDB
Connection: Keep-Alive
Content-Type: text/plain; charset=utf-8
Content-Length: 0

```
The next call will **fetch the job's result** from the server. This will return a 404 error, but the response will contain the `x-arango-async-id` header so the client can tell that this is the original job's result:
```
curl -X PUT http://127.0.0.1:8529/_api/job/147185081573075 --dump - --basic --user "root:"
HTTP/1.1 404 Not Found
X-Arango-Async-Id: 147185081573075
Content-Type: application/json; charset=utf-8
Server: ArangoDB
Connection: Keep-Alive
Content-Length: 94

{"error":true,"code":404,"errorNum":1203,"errorMessage":"unknown collection 'does-not-exist'"}
```
Next I'll be asking the server for the same job's result, which will not be present anymore (because already fetched before). The server will again respond with a 404 error, but will omit the `x-arango-async-id` header now. So the client can tell that this is not the original job's result but an error produced by the job API itself:
```
curl -X PUT http://127.0.0.1:8529/_api/job/147185081573075 --dump - --basic --user "root:"
HTTP/1.1 404 Not Found
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 67

{"error":true,"errorMessage":"not found","code":404,"errorNum":404}
```
In summary, I think it should be possible to tell two 404s apart by looking for the presence of the `x-arango-async-id` header.

**Comment by joowani** [comment]
Hi Jan,

Yes this seems to be exactly what I need. I saw that flag in the documentation but it wasn't obvious  I could use this as the indicator.

Thanks!
Joohwan
